presentation/src/main/java/me/androidbox/presentation/alarm_manager/AlarmReminderProvider.kt

```diff
@@ -0,0 +1,39 @@
+package me.androidbox.presentation.alarm_manager
+
+import me.androidbox.component.agenda.AlarmReminderItem
+import java.time.ZoneId
+import java.time.ZonedDateTime
+
+object AlarmReminderProvider {
+
+    fun getRemindAt(
+        alarmReminderItem: AlarmReminderItem,
+        startTime: ZonedDateTime,
+        startDate: ZonedDateTime
+    ): ZonedDateTime {
+        /** Based on what alarm reminder the user has selected minus the time/date accordingly */
+        return when (alarmReminderItem) {
+            AlarmReminderItem.TEN_MINUTES -> {
+                startTime.minusMinutes(10L)
+            }
+            AlarmReminderItem.THIRTY_MINUTES -> {
+                startTime.minusMinutes(30L)
+            }
+            AlarmReminderItem.ONE_HOUR -> {
+                startTime.minusHours(1L)
+            }
+            AlarmReminderItem.SIX_HOUR -> {
+                startTime.minusHours(6L)
+            }
+            AlarmReminderItem.ONE_DAY -> {
+                startDate.minusDays(1L)
+            }
+        }
+    }
```

Comment:
This could also simply be a member function of `AlarmReminderItem`:
```
enum class AlarmReminderItem {
    TEN_MINUTES, ..., ONE_DAY;

    fun getRemindAt(startTime: ZonedDateTime): ZonedDateTime {
        // Your when expression here
    }
}

// Then called like this:
val remindAt = reminderItem.getRemindAt(startTime)
```
I find this a bit more readable than a separate class which you would not need in that case.